File: solutions/python/robot-name/1/robot_name.py
import random
import string

class Robot:
    names = set()

    def __init__(self):
        self.name = self.generate_name()

    def generate_name(self):
        name = ''.join(random.choices(string.ascii_uppercase, k=2)) + ''.join(random.choices(string.digits, k=3))
        while name in Robot.names:
            name = ''.join(random.choices(string.ascii_uppercase, k=2)) + ''.join(random.choices(string.digits, k=3))
        Robot.names.add(name) 
        return name

    def reset(self):
        self.name = self.generate_name()  

File: solutions/javascript/robot-name/1/robot-name.js
// This is only a SKELETON file for the 'Robot Name' exercise. It's been
// provided as a convenience to get your started writing code faster.

const generateName = () => {
  const letters = String.fromCharCode(
    65 + Math.floor(Math.random() * 26), 
    65 + Math.floor(Math.random() * 26)
  );

  const numbers = Math.floor(Math.random() * 1000)
    .toString()
    .padStart(3, "0");

  const name = letters + numbers;

  if(Robot.namesSet.has(name))
    return generateName();
  
  Robot.namesSet.add(name);
  return name;
}

export class Robot {
  constructor() {
    this._name = generateName();
  }

  get name(){
    return this._name;
  }

  reset() {
    this._name = generateName();
  }

  static releaseNames() {
    Robot.namesSet.clear();
  }
}

Robot.namesSet = new Set();
